src/NuGet.Core/NuGet.Configuration/Settings/ISettings.cs

```diff
@@ -157,5 +157,15 @@ public interface ISettings
         /// Event raised when the setting have been changed.
         /// </summary>
         event EventHandler SettingsChanged;
+
+        /// <summary>
+        /// Get a list of all the paths of the settings files used as part of this settings object
+        /// </summary>
+        IEnumerable<string> GetConfigFilePaths();
```

Comment:
You do a lot of ToList and AsList.

Why not just make this an IList? 
There's no benefit to IEnumerable and creates less garbage.